pkg/api/v1alpha1/versionskew.go

```diff
@@ -0,0 +1,28 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/version"
+)
+
+// SupportedMinorVersionIncrement represents the minor version skew for kubernets version upgrades.
+const SupportedMinorVersionIncrement = 1
+
+// ValidateVersionSkew validates Kubernetes version skew between valid non-nil versions.
+func ValidateVersionSkew(oldVersion, newVersion *version.Version) error {
+	if newVersion.LessThan(oldVersion) {
+		return fmt.Errorf("kubernetes version downgrade is not supported (%s) -> (%s)", oldVersion, newVersion)
+	}
+
+	newVersionMinor := newVersion.Minor()
+	oldVersionMinor := oldVersion.Minor()
+
+	minorVersionDifference := int(newVersionMinor) - int(oldVersionMinor)
+
+	if minorVersionDifference != SupportedMinorVersionIncrement && minorVersionDifference != 0 {
```

Comment:
shouldn't the `minorVersionDifference` <= `SupportedMinorVersionIncrement`
The check works in this case because `SupportedMinorVersionIncrement` is 1 but if this was changed, it would be nice to have that handled as well. Something like this:
```suggestion
	if minorVersionDifference < 0 || minorVersionDifference > SupportedMinorVersionIncrement {
```